gr-digital/lib/crc16_async_bb_impl.cc

```diff
@@ -0,0 +1,105 @@
+/* -*- c++ -*- */
+/*
+ * Copyright 2021 Cesar Martinez.
+ *
+ * This file is part of GNU Radio
+ *
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ * 
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "crc16_async_bb_impl.h"
+#include <gnuradio/io_signature.h>
+#include <volk/volk.h>
+
+namespace gr {
+namespace digital {
+
+crc16_async_bb::sptr crc16_async_bb::make(bool check)
+{
+    return gnuradio::make_block_sptr<crc16_async_bb_impl>(check);
+}
+
+crc16_async_bb_impl::crc16_async_bb_impl(bool check)
+    : block("crc16_async_bb", io_signature::make(0, 0, 0), io_signature::make(0, 0, 0)),
+      d_npass(0),
+      d_nfail(0)
+{
+    d_in_port = pmt::mp("in");
+    d_out_port = pmt::mp("out");
+
+    message_port_register_in(d_in_port);
+    message_port_register_out(d_out_port);
+
+    if (check)
+        set_msg_handler(d_in_port, [this](pmt::pmt_t msg) { this->check(msg); });
+    else
+        set_msg_handler(d_in_port, [this](pmt::pmt_t msg) { this->calc(msg); });
+}
+
+crc16_async_bb_impl::~crc16_async_bb_impl() {}
+
+void crc16_async_bb_impl::calc(pmt::pmt_t msg)
+{
+    // extract input pdu
+    pmt::pmt_t meta(pmt::car(msg));
+    pmt::pmt_t bytes(pmt::cdr(msg));
+
+    unsigned int crc;
```

Comment:
Technically, this should hold at least 16 bits, in order to hold a CRC-16. For me it would be a good idea to use `uint16_t`, since it even helps convey that this is a CRC-16. However, how about instead of declaring `crc` here, doing
```
auto crc = d_crc_ccitt_impl();
```
below and letting the return type of `d_crc_ccitt_impl()` handle this?